tests/test_connection.py

```diff
@@ -34,6 +35,22 @@
 from smbprotocol.session import Session
 
 
+@pytest.fixture
+def connected_session(smb_real):
+    """
+    Return a (connection, session) tuple that is already connected to the
+    SMB server and that will disconnect at the end.
+    """
+    connection = Connection(uuid.uuid4(), smb_real[2], smb_real[3])
+    connection.connect()
+    session = Session(connection, smb_real[0], smb_real[1])
+    session.connect()
+
+    yield connection, session
+
+    connection.disconnect(True)
```

Comment:
Might be best to wrap this in a try/finally

```suggestion
    try:
        yield connection, session
    finally:
        connection.disconnect(True)
```

Reply:
I am not familiar with pytest.

My understanding is that anything after yield is automatically a teardown...so finally is not needed.

https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#handling-errors-for-yield-fixture

Reply:
Ah nice to know, I always thought you had to handle the exceptions but a quick test shows that's not the case.

```python
import pytest


@pytest.fixture
def my_fixture():
    print("Fixture start")

    yield 1

    print("Fixture end")


def test_example(my_fixture):
    assert False
```

```bash
$ python -m pytest /tmp/test_example.py
```

```
====================================================================================== FAILURES ======================================================================================
____________________________________________________________________________________ test_example ____________________________________________________________________________________

my_fixture = 1

    def test_example(my_fixture):
>       assert False
E       assert False

/tmp/test_example.py:14: AssertionError
------------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------------
Fixture start
------------------------------------------------------------------------------ Captured stdout teardown ------------------------------------------------------------------------------
Fixture end
```